charmhelpers/contrib/openstack/context.py

```diff
@@ -352,10 +352,101 @@ def _setup_pki_cache(self):
             return cachedir
         return None
 
+    def _get_ubuntu_pkg_name(self, python_name):
+        """
+        We assume Python3 is the main interpreter and python3-keystonemiddleware
+        is the main package for the transitional period. python-keystonemiddle is
+        here to support possible future
+        (Python 3 and python3-keystonemiddleware->python-keystonemiddleware ?)
+        and the past
+        (Python 2 and python-keystonemiddleware)
+        """
+
+        pkg = 'python3-'
+        if filter_installed_packages((pkg + python_name,)):
+            pkg = 'python-' + python_name
+
+        return pkg
+
+    def _get_keystone_authtoken_ctxt(self, ctxt, keystonemiddleware_os_rel):
+        """
+        Re-constructed from former template 'section-keystone-auth-mitaka'
+        """
+        config_items = []
+
+        if CompareOpenStackReleases(keystonemiddleware_os_rel) >= 'mitaka':
```

Comment:
I am *really* sorry to mess you about. I know I suggested we should have support from the earliest release but I think I was wrong. I think we should support from Queens and above only.

Reply:
No problem. This part transforms just to:
```
config_items.append(
    ('auth_type', 'password')
)
```
then. Also given that input, i think it is might be safe to drop charmhelpers/contrib/openstack/section-keystone-authtoken altogether, but not sure...